### ISSUE TITLE
Logs flooded with warning message - Detected ES 7.x or above: `_doc` will be used as the document `_type`

### ISSUE THREAD

**Issue by nvenky** [uken/fluent-plugin-elasticsearch#589]
#### Problem

Fluentd Logs is flooded with the following log message.
[elasticsearch] Detected ES 7.x or above: `_doc` will be used as the document `_type`

#### Steps to replicate

We are using Fluentd in Kubernetes environment. We used the Helm stable/fluentd-elasticsearch chart to deploy.

Custom output config with json parsing enabled:
```aconf
      # Enriches records with Kubernetes metadata
      <filter kubernetes.**>
        @type kubernetes_metadata
      </filter>

      #https://github.com/fluent/fluentd/issues/2073#issuecomment-405503834
      <filter kubernetes.var.log.containers.**>
        @type parser
        <parse>
          @type json
          json_parser json
        </parse>
        replace_invalid_sequence true
        emit_invalid_record_to_error false
        key_name log
        reserve_data true
      </filter>

      <match **>
        @id elasticsearch
        @type elasticsearch
        @log_level info
        include_tag_key true
        type_name _doc
        host "#{ENV['OUTPUT_HOST']}"
        port "#{ENV['OUTPUT_PORT']}"
        scheme "#{ENV['OUTPUT_SCHEME']}"
        ssl_version "#{ENV['OUTPUT_SSL_VERSION']}"
        logstash_format true
        # https://github.com/atomita/fluent-plugin-aws-elasticsearch-service/issues/15#issuecomment-254793259
        reload_on_failure false
        reload_connections false
        reconnect_on_error true
        logstash_prefix "#{ENV['LOGSTASH_PREFIX']}"

        <buffer>
          @type file
          path /var/log/fluentd-buffers/kubernetes.system.buffer
          flush_mode interval
          retry_type exponential_backoff
          flush_thread_count 2
          flush_interval 5s
          retry_forever
          retry_max_interval 30
          chunk_limit_size "#{ENV['OUTPUT_BUFFER_CHUNK_LIMIT']}"
          queue_limit_length "#{ENV['OUTPUT_BUFFER_QUEUE_LIMIT']}"
          overflow_action block
        </buffer>
      </match>
```

#### Expected Behavior or What you need to ask
Maybe log the warning once, but not flood the whole log file.
...

#### Using Fluentd and ES plugin versions

* OS version
* Bare Metal or within Docker or Kubernetes or others?
Kubernetes environment.
* Fluentd v0.12 or v0.14/v1.0
  * fluentd 1.2.4
* ES plugin 3.x.y/2.x.y or 1.x.y
  * paste boot log of fluentd or td-agent

`
  activesupport (5.2.1)
  addressable (2.5.2)
  bigdecimal (1.2.8)
  concurrent-ruby (1.1.3)
  cool.io (1.5.3)
  did_you_mean (1.0.0)
  dig_rb (1.0.1)
  domain_name (0.5.20180417)
  elasticsearch (6.1.0)
  elasticsearch-api (6.1.0)
  elasticsearch-transport (6.1.0)
  excon (0.62.0)
  faraday (0.15.3)
  ffi (1.9.25)
  fluent-plugin-concat (2.3.0)
  fluent-plugin-detect-exceptions (0.0.11)
  fluent-plugin-elasticsearch (2.11.11)
  fluent-plugin-kubernetes_metadata_filter (2.0.0)
  fluent-plugin-multi-format-parser (1.0.0)
  fluent-plugin-prometheus (1.0.1)
  fluent-plugin-systemd (1.0.1)
  fluentd (1.2.4)
  http (0.9.8)
  http-cookie (1.0.3)
  http-form_data (1.0.3)
  http_parser.rb (0.6.0)
  i18n (1.1.1)
  io-console (0.4.5)
  json (1.8.3)
  kubeclient (1.1.4)
  lru_redux (1.1.0)
  mime-types (3.2.2)
  mime-types-data (3.2018.0812)
  minitest (5.11.3, 5.9.0)
  msgpack (1.2.4)
  multi_json (1.13.1)
  multipart-post (2.0.0)
  net-telnet (0.1.1)
  netrc (0.11.0)
  oj (3.6.13)
  power_assert (0.2.7)
  prometheus-client (0.8.0)
  psych (2.1.0)
  public_suffix (3.0.3)
  quantile (0.2.1)
  rake (10.5.0)
  rdoc (4.2.1)
  recursive-open-struct (1.0.0)
  rest-client (2.0.2)
  serverengine (2.0.7)
  sigdump (0.2.4)
  strptime (0.2.3)
  systemd-journal (1.3.3)
  test-unit (3.1.7)
  thread_safe (0.3.6)
  tzinfo (1.2.5)
  tzinfo-data (1.2018.7)
  unf (0.1.4)
  unf_ext (0.0.7.5)
  yajl-ruby (1.4.1)
`
* ES version (optional)
  Elasticsearch 7 - Docker instance.
* ES template(s) (optional)

**Comment by cosmo0920** [comment]
Could you update fluent-plugin-elasticsearch v3.5.0 or later?

**Comment by jnicholls** [comment]
Also experiencing this issue, even though I provided `_doc` as the type_name per the OP. I think I will try to get fluentd to run in quiet mode until it can be upgraded (I'm using Rancher and they need to update their fluentd helm chart).

**Comment by cosmo0920** [comment]
This issue had been fixed in v3.4.2:
https://github.com/uken/fluent-plugin-elasticsearch/pull/573
You should use v3.4.2 or later.
